packages/shorebird_code_push_client/lib/src/code_push_client.dart

```diff
@@ -86,6 +189,7 @@ class CodePushClient {
     http.Client? httpClient,
     Uri? hostedUri,
   })  : _httpClient = _CodePushHttpClient(httpClient ?? http.Client()),
+        _uploadProgressClient = _UploadProgressHttpClient(),
```

Comment:
Seems weird to have to inject a separate client for the uploads. If we go with this approach it might be nicer to just have a `ShorebirdHttpClient` which includes the retry logic, logging, and optional progress reporting.

Reply:
Yeah, I agree. I just explored moving this into the `shorebird_cli` package, but was thwarted because 1) this wraps a `dart:io` HttpClient, not a `package:http` `http.Client`, meaning it has to be the innermost wrapped class (with LoggingClient and RetryingClient around it), which in turn means we don't have a ready way to access the APIs it offers that aren't on `http.BaseClient` and 2) `CodePushClient` needs to be aware of the types that are used in the progress callback. We could do away with the second problem if we just communicated progress through a primitive type, but I don't have a clean solution to the first issue offhand.